std/file.d

```diff
@@ -5119,3 +5120,87 @@ string tempDir() @trusted
     myFile.write("hello");
     assert(myFile.readText == "hello");
 }
+
+/**
+Returns the available space in a file system.
+
+On `Windows`, this function uses the result of the Windows API function
+$(LINK2 https://msdn.microsoft.com/en-us/library/windows/desktop/aa364937(v=vs.85).aspx, $(D GetDiskFreeSpaceExW)).
+
+On `POSIX` platforms, it uses the $(LINK2 http://man7.org/linux/man-pages/man3/statvfs.3.html,
+$(D statvfs)) function. On `FreeBSD` however, it uses the $(LINK2 https://www.freebsd.org/cgi/man.cgi?query=statfs&sektion=2,
+$(D statfs)) function.
```

Comment:
I still think that the user doesn't care about this. He cares about the _implications_ of using these APIs.
If there's something different on Windows, note this, but otherwise the users expects that we have done our job properly.

Reply:
https://www.boost.org/doc/libs/1_48_0/libs/filesystem/v3/doc/reference.html#space
Boost explains very clearly what happens in their filesystem library. I think it's a good idea to let the user know how we are getting the sizes.

Reply:
As long as this extra information comes "below the fold" I think it's fine to have any amount of extra technical information.

Reply:
Nope it's not. It's our good to provide a "high-level" abstraction (as in the user shouldn't need to care about the low-level).
He definitely doesn't want to read 3 different documentations just to know "hey they return the disk size". If he really wants to, he can always look this up in the sources.

Also have a look at the existing std.file - notice how clean it is?
One of the few function where OS details are mentioned is [tempDir](https://dlang.org/phobos-prerelease/std_file.html#.tempDir) and it matters there.

Reply:
Taking a second look , the "however" makes it seem like there's an important distinction the user needs to be aware of. Is there one?
>On `FreeBSD` however, it uses the statfs function.

Reply:
Here's a good example of an OS abstraction from Python's standard library:

https://docs.python.org/3/library/shutil.html#shutil.disk_usage